model_graph/trie_model/trie_node.py

```diff
@@ -0,0 +1,49 @@
+from typing import Generator, NamedTuple
+
+
+class TitleObj(NamedTuple):
+    title: str
+    score: int
+
+    
+class TrieNode:
+    def __init__(self, char:str, score:int=1, children:dict[str, 'TrieNode']=None):
+        self.children = children if children else {}
+        self.char = char
+        self.score = score
+
+    def insert(self, text):
+        if not text:
+            return
+        char = text[0]
+        if char not in self.children:
+            self.children[char] = TrieNode(char, 1, {})
+        else:
+            self.children[char].score += 1
+
+        self.children[char].insert(text[1:])
+
+    def find_node(self, path:str):
+        if not path:
+            return self 
+        elif path[0] not in self.children:
+            return None
+        return self.children.get(path[0]).find_node(path[1:])
+ 
+    def get_all_titles(self, prefix:str) -> Generator[TitleObj,None,None]:
```

Comment:
I think that is for Iterator, Generator required[YieldType, SendType, ReturnType], otherwise it yields this error.
TypeError: Too few arguments for typing.Generator; actual 1, expected 3